torch/_higher_order_ops/scan.py

```diff
@@ -48,6 +48,13 @@ def _extract_carry_and_out(flat_out: List[Any], num_carry: int):
     return flat_out[:num_carry], flat_out[num_carry:]
 
 
+# an empty function that's used for inductor lowering in lowering.py
+# difference is that this function supports tensor idx while select doesn't
+# dst.select(dim, idx).copy_(src)
+def override_slice(dst, src, dim, idx):
```

Comment:
This is a pretty weird way of defining a custom op. I would just define a custom using the custom op API (cc: @zou3519 ).

See https://github.com/pytorch/pytorch/blob/dd443f418ab8bfd7d30be640d03fd1dc113ebc35/torch/library.py#L648

Reply:
I'm not sure if we should have a legit custom op for it or just keep it as empty. What i want is actually just a special function that's lowered differently from select because we're sure shape checks are not necessary and we need clamp argument to be False). We don't need this operator to work in eager or any dispatch keys i feel.

Reply:
We should make this a custom op (via torch.library.custom_op). It's not difficult (it's like two more lines of code). We might want to add more invariants to inductor in the future, like "you can only define lowerings for operators and HOPs".

Reply:
Yeah, sounds good then. Since we're going with the re-tracing approach for creating the out variant graph, we'll need a fake implementation. Let me add it.